### PR TITLE
chore: add missing 'use client'/'use server' directives

### PR DESCRIPTION
**Pull request by christianhg**

### CODE REVIEW

**Review by stipsan — CHANGES_REQUESTED** [review]
Ah, good catch! Turbopack must've started using the `source` field to optimize HMR when running dev 🙌

Could you update the directives to be in the same locations as where our bundler is adding them? https://github.com/sanity-io/next-sanity/blob/8c5e28bc8223db66028ae5ec5024eadd8ed18cc9/packages/next-sanity/package.config.ts#L5-L13

Unfortunately rollup, which is used under the hood, doesn't support preserving these directives so it'll still be necessary to add them in the bundler even though we're also adding them to the src 😅

**Comment by christianhg** [comment]
> Unfortunately rollup, which is used under the hood, doesn't support preserving these directives so it'll still be necessary to add them in the bundler even though we're also adding them to the src 😅

Good to know! I'll add them here as well.

File: packages/next-sanity/src/studio/client-component/NextStudio.tsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+"use client"
```
Comment (stipsan) [inline packages/next-sanity/src/studio/client-component/NextStudio.tsx:1]:
```suggestion
```
This directive should be lifted to the client-component/index file

**Comment by christianhg** [comment]
> > Unfortunately rollup, which is used under the hood, doesn't support preserving these directives so it'll still be necessary to add them in the bundler even though we're also adding them to the src 😅
> 
> Good to know! I'll add them here as well.

I've now amended the commit to include the new files in that config.

File: packages/next-sanity/src/studio/client-component/registry.tsx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+"use client"
```
Comment (stipsan) [inline packages/next-sanity/src/studio/client-component/registry.tsx:1]:
```suggestion
```
This directive should be lifted to the client-component/index file

File: packages/next-sanity/src/studio/client-component/useIsMounted.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+"use client"
```
Comment (stipsan) [inline packages/next-sanity/src/studio/client-component/useIsMounted.ts:1]:
```suggestion
```
This directive should be lifted to the client-component/index file

**Review by stipsan — CHANGES_REQUESTED** [review]

File: packages/next-sanity/src/studio/client-component/useIsMounted.ts (hunk 3 above)
Comment (christianhg) [reply packages/next-sanity/src/studio/client-component/useIsMounted.ts:1]: Aaah of course. I had the same hunch about maybe lifting these. Will do.

**Comment by stipsan** [comment]
> > > Unfortunately rollup, which is used under the hood, doesn't support preserving these directives so it'll still be necessary to add them in the bundler even though we're also adding them to the src 😅
> 
> > 
> 
> > Good to know! I'll add them here as well.
> 
> 
> 
> I've now amended the commit to include the new files in that config.


Ooops, I meant the opposite ☺️ the files currently declared in package.config.ts is the only locations we should be adding the directive. They need to match 1:1 otherwise the behavior of `next dev` inside this monorepo will behave and bundle wildly different than `next dev` in userland.
It'll also cause the Vercel deployments on this monorepo that uses `next build` to also have use client in different places.
The reason for this is that when rollup bundles it removes all directives it doesn't understand, and currently it only understands `use strict`

**Comment by christianhg** [comment]
> Ooops, I meant the opposite ☺️ the files currently declared in package.config.ts is the only locations we should be adding the directive.

Got it! The PR is now much simpler.

**Comment by stipsan** [comment]
Could you add `use client` to image/index as well? And `use server' to server-actions/index?
That way `pnpm dev` behaves exactly like the code we publish to npm ✌️

**Comment by christianhg** [comment]
> Could you add `use client` to image/index as well? And `use server' to server-actions/index? That way `pnpm dev` behaves exactly like the code we publish to npm ✌️

Will do! Your overview here is impeccable. I apologise for all the hand-holding you need to make.

**Comment by stipsan** [comment]
> > Could you add `use client` to image/index as well? And `use server' to server-actions/index? That way `pnpm dev` behaves exactly like the code we publish to npm ✌️
> 
> 
> 
> Will do! Your overview here is impeccable. I apologise for all the hand-holding you need to make.

No need! Custom directives is novel and bleeding edge, we don't expect new joiners to understand how Next.js internals and React features yet to land in a stable release works within their first month on the team 😉

**Comment by stipsan** [comment]
Could you change the commit prefix to "chore" btw? Since these changes only impact the monorepo and not the code we publish to npm 😌

**Comment by christianhg** [comment]
> Could you change the commit prefix to "chore" btw? Since these changes only impact the monorepo and not the code we publish to npm 😌

Haha, yes, I just thought about the same thing. No need to publish this change. I think I need coffee...

**Review by stipsan — APPROVED** [review]
All good now! 🥳

**Pull request merged**